#228 Re-factor the latest filters

alexstark4 opened the pull request:
* Shortened class names significantly
* split AsymmetricPeakShapeFilter into two filters
* Shortened class names/descriptions
* provided more options for Treatment
* Shortened class names/descriptions for Criterion

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor the latest filters

* Shortened class names significantly
* split AsymmetricPeakShapeFilter into two filters
* Shortened class names/descriptions
* provided more options for Treatment
* Shortened class names/descriptions for Criterion

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 commented:
TODO: add Preference Page(s)

@eselmeister @laeubi What is the most efficient method to implement Preference Page(s)? Alex K. suggested using Json to annotate the settings, but that doesn't create Preference Page(s).

alexstark4 commented:
@kerner1000

kerner1000 commented on chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java:
```diff
@@ -0,0 +1,144 @@
+		for(X peak : read) {
+			double compareAreaValue = calculatePercentageAreaCompareValue(peak, areaSum);
+			applySelectedOptions(compareAreaValue, configuration, listener, peak);
+			checkForKeepOption(listener, configuration, peak);
```
Check the keep option inside the "processPeak" method.

kerner1000 commented on chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java:
```diff
@@ -0,0 +1,144 @@
+			peak.setActiveForAnalysis(false);
+			listener.updated(peak);
+			break;
+		case KEEP_PEAK:
```
I dont understand this. Why do you modify here the active for analysis property?
As far as I can tell, it should be:
if peak matches filter criterion, do nothing. Else, apply "Delete peak" option.

alexstark4 replied:
The idea behind this is that the peak to be kept should be differentiated from the peak to be deleted. Therefore, the property is first modified and then the marked peaks are removed. In this way, the peaks sought are retained.

If I think about it again, this approach is not entirely optimal.

alexstark4 replied:
According to Philip, the requirement is that the peaks that are not kept should also be deleted.

kerner1000 commented:
Next time please do changes such as refactoring of names in a dedicated commit.

alexstark4 committed:
#228 Re-factor peak processing in IntegratedAreaFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 commented on chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/IntegratedAreaFilter.java:
```diff
@@ -63,34 +63,60 @@ public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
+		boolean keepFlag = false;
+		if(configuration.getFilterTreatmentOption()==ValueFilterTreatmentOption.KEEP_PEAK) {
+			keepFlag = true;
+		}
```
@kerner1000 Is it easier to understand? The flag is used to virtually invert the deletion of the peaks.

alexstark4 committed:
#228 Re-factor peak processing in PercentageAreaFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor peak processing in PeakShapeFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor peak processing in PeakAsymmetryFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor javadocs of the enums

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

kerner1000 committed:
Refactor to work without a 'keepFlag'

kerner1000 commented:
See https://github.com/alexstark4/chemclipse/pull/1 for some suggestions.

kerner1000 committed:
Simplify filter name

If a peak area was obtained by an integration or anything else is not
fixed and not important in this context.

kerner1000 committed:
Minor tweak

alexstark4 committed:
Merge pull request #1 from kerner1000/refactor-latest-filters_

Refactor to work without a 'keepFlag'

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>
Change-Id: Ie4059c97834a911acca142048ab9e2b49b5c3dd0

alexstark4 committed:
Merge pull request #1 from kerner1000/refactor-latest-filters_

Refactor to work without a 'keepFlag'

alexstark4 committed:
#228 Further shorten filter names

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor AreaPercentFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Optimize getName() return value

alexstark4 committed:
#228 Optimize getName() return value

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
Merge branch 'refactor-latest-filters_' of github.com:alexstark4/chemclipse into refactor-latest-filters_

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>
Change-Id: I373361b9ba83d47dc80a25a4c724398cacf2b12a

alexstark4 committed:
#228 Re-factor AsymmetryFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
#228 Re-factor ShapeFilter

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

alexstark4 committed:
Merge branch 'develop' into refactor-latest-filters_

alexstark4 committed:
Merge branch 'refactor-latest-filters_' of github.com:alexstark4/chemclipse into refactor-latest-filters_

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>
Change-Id: I07ad33d439ed39a5eb24e60b95f3b4ea97371cc6

alexstark4 committed:
#228 Add S/N Filter in MANIFEST.MF after merging

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

laeubi commented:
Please reopen with suitable ECA
